src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -42,19 +52,47 @@ internal abstract class AbstractSnippetExpansionClient : ForegroundThreadAffinit
         protected readonly ITextView TextView;
         protected readonly ITextBuffer SubjectBuffer;
 
+        private readonly IEnumerable<Lazy<ArgumentProvider, OrderableLanguageMetadata>> _allArgumentProviders;
+        private ImmutableArray<ArgumentProvider> _argumentProviders;
+
         protected bool indentCaretOnCommit;
         protected int indentDepth;
         protected bool earlyEndExpansionHappened;
 
         internal IVsExpansionSession ExpansionSession;
-
-        public AbstractSnippetExpansionClient(IThreadingContext threadingContext, Guid languageServiceGuid, ITextView textView, ITextBuffer subjectBuffer, IVsEditorAdaptersFactoryService editorAdaptersFactoryService)
+        private bool _preserveSymbols;
+        private ImmutableArray<IMethodSymbol> _symbols;
+        private IMethodSymbol _symbol;
+        private ImmutableDictionary<string, string> _arguments = ImmutableDictionary.Create<string, string>();
```

Comment:
it's not immediately evident what these fields are for.  consider doc'ing

Reply:
⏱️ Added docs as part of state extraction